src/ripple/protocol/impl/STAmount.cpp

```diff
@@ -1179,24 +1207,13 @@ mulRound (STAmount const& v1, STAmount const& v2, Issue const& issue,
         }
     }
 
-    bool resultNegative = v1.negative() != v2.negative();
-    // Compute (numerator * denominator) / 10^14 with rounding
-    // 10^16 <= result <= 10^18
-    CBigNum v;
-
-    if ((BN_add_word64 (&v, value1) != 1) || (BN_mul_word64 (&v, value2) != 1))
-        Throw<std::runtime_error> ("internal bn error");
-
-    if (resultNegative != roundUp) // rounding down is automatic when we divide
-        BN_add_word64 (&v, tenTo14m1);
+    bool const resultNegative = v1.negative() != v2.negative();
 
-    if  (BN_div_word64 (&v, tenTo14) != 1)
-        Throw<std::runtime_error> ("internal bn error");
-
-    // 10^16 <= product <= 10^18
-    assert (BN_num_bytes (&v) <= 64);
+    // Compute (numerator * denominator) / 10^14 with rounding
+    std::uint64_t amount = fast_muldiv (
+        value1, value2, tenTo14,
+        (resultNegative != roundUp) ? tenTo14m1 : 0);
```

Comment:
If I'm reading this code correctly, it actually functions as round-away-from-zero, not round up.  If I'm correct that's fine, but it should be documented that way, which would mean changing the name of the function argument.

If the operation really is round-away-from-zero here, do `canonicalizeRound()` and `divRound()` also round away from zero?  If so their argument names might need adjusting as well.  If not, then why would we not want all three to have the same rounding behavior?